app/atomic_svc.py

```diff
@@ -0,0 +1,240 @@
+
+        if not os.path.exists(self.repo_dir) or not os.listdir(self.repo_dir):
+            check_call(['git', 'clone', '--depth', '1', repo_url, self.repo_dir], stdout=DEVNULL, stderr=STDOUT)
+
+    async def populate_data_directory(self, path_yaml=None):
+        '''
+        Populate the 'data' directory with the Atomic Red Team abilities.
+        These data will be usable by caldera after importation.
+        You can specify where the yaml files to import are located with the `path_yaml` parameter.
+        By default, read the yaml files in the atomics/ directory inside the Atomic Red Team repository.
+        '''
+        if not self.technique_to_tactics:
+            await self._populate_dict_techniques_tactics()
+
+        if not path_yaml:
+            path_yaml = os.path.join(self.repo_dir, 'atomics', '**', 'T*.yaml')
+
+        for filename in glob.iglob(path_yaml):
+            for entries in BaseWorld.strip_yml(filename):
+                for test in entries['atomic_tests']:
+                    try:
+                        incr_ingested, incr_total = await self._save_ability(entries, test)
+                        self.at_ingested += incr_ingested
+                        self.at_total += incr_total
+                    except Exception as e:
+                        self.log.debug('ERROR:', filename, e)
+                        self.errors += 1
+
+        errors_output = f' and ran into {self.errors} errors' if self.errors else ''
+        self.log.debug(f'Ingested {self.at_ingested} abilities (out of {self.at_total}) from Atomic plugin{errors_output}')
+
+    """ PRIVATE """
+
+    def _gen_single_match_tactic_technique(self, mitre_json):
+        '''
+        Generator parsing the json from 'enterprise-attack.json',
+        and returning couples (phase_name, external_id)
+        '''
+        for obj in mitre_json.get('objects'):
+            if not obj.get('type') == 'attack-pattern':
+                continue
+            for e in obj.get('external_references'):
+                if not e.get('source_name') == 'mitre-attack':
+                    continue
+                external_id = e.get('external_id')
+                for kc in obj.get('kill_chain_phases'):
+                    if not kc.get('kill_chain_name') == 'mitre-attack':
+                        continue
+                    phase_name = kc.get('phase_name')
+                    yield (phase_name, external_id)
+
+    async def _populate_dict_techniques_tactics(self, entreprise_attack_path=None):
+        '''
+        Populate internal dictionary used to match techniques to corresponding tactics.
+        By default, use the file 'enterprise-attack.json' located in the Atomic Red Team repository.
+        You can use a specific version of this file by passing its path with the `entreprise_attack_path`
+        parameter.
+        '''
+        self.technique_to_tactics = defaultdict(list)
+        if not entreprise_attack_path:
```

Comment:
we can remove this if check, as we control (line 98) what is passed in, and our only case is to pass in None

Reply:
fixed in 5b66cb5, also corrected the doc